### PR TITLE
Documentation GC

### PR DESCRIPTION
**Pull request by juancarlospaco**
- Documentation only.
- Follow up from https://github.com/nim-lang/Nim/pull/14609/files
:)

### CODE REVIEW

**Comment by Araq** [comment]
"Stop-The-World" is maybe not precise enough. Nim's markAndSweep GC, for instance, does stop the thread it runs on.

**Comment by juancarlospaco** [comment]
@Araq 
`stop-the-world garbage collectors completely halt execution of the program to run a collection cycle`
https://en.wikipedia.org/wiki/Tracing_garbage_collection#Stop-the-world_vs._incremental_vs._concurrent

Do you suggest a better naming?. Been asked this few times by people.

File: doc/gc.rst [hunk 1]
```diff
@@ -43,17 +43,17 @@ To choose the memory management strategy use the ``--gc:`` switch.
...
-RefC               Local    Cycle Collector   ``--gc:refc``
-Mark & Sweep       Local    Cycle Collector   ``--gc:markAndSweep``
-ARC                Shared   Leak              ``--gc:arc``
-ORC                Shared   Cycle Collector   ``--gc:orc``
-Boehm              Shared   Cycle Collector   ``--gc:boehm``
-Go                 Shared   Cycle Collector   ``--gc:go``
-None               Manual   Manual            ``--gc:none``
-================== ======== ================= ======================
+================== ======== ================= ============== ===================
+Memory Management  Heap     Reference Cycles  Stop-The-World Command line switch
```
Comment (Varriount) [inline doc/gc.rst:47]: What does "stop the world" mean in this context? One thread, all threads, deterministic vs indeterministic, etc?

**Review by Varriount — CHANGES_REQUESTED** [review]

File: doc/gc.rst (hunk 1 above)
Comment (juancarlospaco) [reply doc/gc.rst:47]: Unless explicitly specified people usually refer to all the threads of a process. Do you suggest a better naming?.

**Pull request merged**